tool/tsh/proxy.go

```diff
@@ -32,26 +34,43 @@ import (
 )
 
 func onProxyCommandSSH(cf *CLIConf) error {
-	client, err := makeClient(cf, false)
+	tc, err := makeClient(cf, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	address, err := utils.ParseAddr(tc.WebProxyAddr)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	address, err := utils.ParseAddr(client.WebProxyAddr)
+	key, err := tc.LocalAgent().GetKey(tc.SiteName)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
+	pool := x509.NewCertPool()
+	for _, caPEM := range key.TLSCAs() {
+		if !pool.AppendCertsFromPEM(caPEM) {
+			return trace.BadParameter("failed to parse TLS CA certificate")
+		}
+	}
```

Comment:
Nit: Might be worth it to tuck this into a helper method on the agent so you could do something like

```go
pool, err := tc.LocalAgent().ClientCertPool(tc.SiteName)
```